stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/PushMessageSyncHandler.kt

```diff
@@ -0,0 +1,171 @@
+package io.getstream.chat.android.livedata.service.sync
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.notifications.FirebaseMessageParser
+import io.getstream.chat.android.client.notifications.FirebaseMessageParserImpl
+import io.getstream.chat.android.client.notifications.handler.ChatNotificationHandler
+import io.getstream.chat.android.client.notifications.handler.NotificationConfig
+import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
+import io.getstream.chat.android.livedata.ChatDomain
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+/**
+ * A delegate class which can be used in [com.google.firebase.messaging.FirebaseMessagingService] in case the app uses
+ * multiple push notifications backend services.
+ *
+ * If the app uses Stream service as its only push notifications provider, the [OfflineSyncFirebaseMessagingService]
+ * class can be used as the  out-of-the box implementation instead of [PushMessageSyncHandler] methods.
+ */
+public class PushMessageSyncHandler(private val service: Service) {
```

Comment:
Passing in this `service` instance makes for easy integration (as in few lines) for the client, but I'm really not sure if they'll expect us to stop/start this service for them.

Reply:
Do you mean starting the service foreground execution and then stoping it after sync? 
```kotlin 
service.startForeground(NOTIFICATION_ID, this)
```
and
```kotlin
service.stopForeground(true)
```
Why is it bad?

Reply:
I think they could have code that would run when the service is started or stopped, leading to potential side effects. Haven't worked with Services in a while though, just feels like we should be careful around these parts.